src/test/java/tests/LoginTest.java

```diff
@@ -35,4 +36,53 @@ public void TC002() {
 
         Assert.assertEquals(actualErrorMsg, expectedErrorMsg, "Error message is incorrect.");
     }
+
+    @Test(description = "User cannot log into Railway with invalid password")
+    public void TC003() {
+        homePage.gotoLoginPage();
+
+        String invalidPassword = DataHelper.getRandomText();
+
+        loginPage.login(Constant.USERNAME, invalidPassword);
+
+        String expectedErrorMsg = "There was a problem with your login and/or errors exist in your form.";
+        String actualErrorMsg = loginPage.getErrorMsg();
+
+        Assert.assertEquals(actualErrorMsg, expectedErrorMsg, "Error message is incorrect.");
+        Assert.assertEquals(actualErrorMsg, expectedErrorMsg, "Welcome message is incorrect.");
+    }
+
+    @Test(description = "Login page displays when un-logged User clicks on Book ticket tab")
+    public void TC004() {
+        homePage.gotoBookTicketPage();
+
+        String expectedPageTitle = "Safe Railway - Login";
+        String actualPageTitle = BrowserHelper.getPageTitle();
+
+        Assert.assertEquals(actualPageTitle, expectedPageTitle, "Login page doesn't display");
+    }
+
+    @Test(description = "Additional pages display once user logged in")
+    public void TC006() {
+        homePage.gotoLoginPage();
+        loginPage.login(Constant.USERNAME, Constant.PASSWORD);
+
+        Assert.assertTrue(homePage.doesTabMyTicketExist());
+        Assert.assertTrue(homePage.doesTabLogoutExist());
+        Assert.assertTrue(homePage.doesChangePasswordExist());
+
+        homePage.gotoMyTicketPage();
+
+        String expectedMyTicketPageTitle = "Safe Railway - My Ticket";
+        String actualPageTitle = BrowserHelper.getPageTitle();
+
+        Assert.assertEquals(actualPageTitle, expectedMyTicketPageTitle, "My Ticket page doesn't display");
+
+        homePage.gotoChangePasswordPage();
+
+        String expectedChangePasswordPageTitle = "Safe Railway - Change Password";
+        actualPageTitle = BrowserHelper.getPageTitle();
```

Comment:
same as above

Reply:
Thanks for your review. I just have updated my code, please review it for me.